Adding new design issues template

liaprins-czi opened the pull request:
# Description
<!-- What does this pull request (PR) do? Why is it necessary? -->
<!-- Tell us about your new feature, improvement, or fix! -->
<!-- If your change includes user interface changes, please add an image, or an animation "An image is worth a thousand words!" -->
<!-- You can use https://www.cockos.com/licecap/ or similar to create animations -->

## Type of change
<!-- Please delete options that are not relevant. -->
- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
<!-- What resources, documentation, and guides were used in the creation of this PR? -->
<!-- If this is a bug-fix or otherwise resolves an issue, reference it here with "closes #(issue)" -->

# How has this been tested?
<!-- Please describe the tests that you ran to verify your changes. -->
- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
_Not testing, but it is just a template so hopefully that's okay?_

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

liaprins-czi committed:
Update issue templates

liaprins-czi committed:
Merge pull request #2 from liaprins-czi/liaprins-czi-design-template

Add design issue template

liaprins-czi committed:
Merge branch 'napari:master' into master

liaprins-czi committed:
Delete --design-related.md

liaprins-czi committed:
Merge pull request #3 from liaprins-czi/liaprins-czi-removing-design-template

Delete --design-related.md

liaprins-czi committed:
Update issue templates

sofroniewn commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -1,7 +1,10 @@
 about: Submit a proposal/request for a new napari feature
-labels: feature
+title: ''
+labels: ''
```
```suggestion
labels: feature
```

sofroniewn replied:
not sure why we lost this

sofroniewn commented on .github/ISSUE_TEMPLATE/task.md:
```diff
@@ -1,7 +1,10 @@
 about: Submit a proposal/request for a new napari feature
-labels: task
+title: ''
+labels: ''
```
```suggestion
labels: task
```

sofroniewn replied:
not sure why we lost this

sofroniewn commented on .github/ISSUE_TEMPLATE/--design-related.md:
```diff
@@ -0,0 +1,38 @@
+---
```
Can we drop the `--` from the beginning of this file name

liaprins-czi replied:
Hmm that's weird, I think it is there because I added an emoji to the title, to matke it match the other templates that had emojis at the front, and it must have converted the emoji + space to dashes.

jni replied:
If you look at the other files, they have the emoji in the name inside the table but not in the filename, so I think the filename essentially doesn't matter.

The other files have the emoji in the name entry and the title is blank — I don't know how that difference works but I thought it might be worth flagging.

sofroniewn replied:
huh, weird. I'm also not so bothered by this, but it is funny. i might try deleting `--` and seeing what happens, if it all works fine without then i'd prefer not to have for consistency reasons

sofroniewn commented on .github/ISSUE_TEMPLATE/--design-related.md:
```diff
@@ -0,0 +1,38 @@
+- [ ] **No:** engineering can create a first version, and design can come in later to iterate and refine
+
+If selecting **Yes**, how much design input is needed to unblock engineering? For example, is a full, final visual design needed, or just a recommendation of which conceptual direction to go?
+
```
Can we maybe put a section break here i.e. a `----------` or something else to indicate that this section is not to be completed up the submitter.

liaprins-czi replied:
Yeah, I am actually removing this part from the template, I will include it as a comment for any issues made with this template, to be less confusing (and since I can't edit the main issue text made by someone else anyway!)

sofroniewn reviewed (approved):
Looks good to me! I've left some minor comments. Maybe @jni can review this too before merge, especially with an eye to making sure the language is accessible to new comers on the repo.

kne42 commented on .github/ISSUE_TEMPLATE/--design-related.md:
```diff
@@ -0,0 +1,38 @@
+- [ ] ●●○○○ S: able to do within one sprint, if also working on other things in the sprint
+- [ ] ●●●○○ M: able to do within one sprint, if it is the sole focus
+- [ ] ●●●●○ L: will likely take more than one sprint to complete, but less than two
+- [ ] ●●●●● XL: a very large experience to be determined or overhauled, will need to be broken down further
```
since the person submitting this doesn't choose these options, maybe it would be better to just create labels for these instead?

liaprins-czi replied:
Yes great point; it probably doesn't make sense as part of the template

kne42 reviewed (approved):
Love the template!

liaprins-czi committed:
Updates to design.md issue template

Updates:
- Hopefully fixed the name having `--` at the front, by removing title's preceding emoji + space
- Removed "T-shirt sizing" section and will instead include that section as a comment for any issues made with this template, to be less confusing (and since I can't edit the main issue text made by someone else anyway!)
- Updated mentions of "research" to be more clearly about UX research, per Lucy's suggestions

liaprins-czi committed:
Update .github/ISSUE_TEMPLATE/feature_request.md

Co-authored-by: Nicholas Sofroniew <sofroniewn@gmail.com>

liaprins-czi committed:
Update .github/ISSUE_TEMPLATE/task.md

Co-authored-by: Nicholas Sofroniew <sofroniewn@gmail.com>

sofroniewn committed:
Rename --design-related.md to design_related.md
